server/service/hosts.go

```diff
@@ -927,10 +927,11 @@ func (svc *Service) OSVersions(ctx context.Context, teamID *uint, platform *stri
 	}
 
 	osVersions, err := svc.ds.OSVersions(ctx, teamID, platform)
-	if err != nil {
+	if err != nil && fleet.IsNotFound(err) {
+		osVersions = &fleet.OSVersions{}
```

Comment:
Maybe check if the team actually exists before returning an empty array.

Reply:
Good idea.